### DISCUSSION TITLE
Getting "true" for multiple checkboxes

### DISCUSSION THREAD

**Discussion by simplenotezy** [react-hook-form/react-hook-form#5762] [Q&A] [answered]
I am playing around with multiple checkboxes, and for some weird issue it keeps returning `true` for everything I do. I would expect:

    chooseTwo: ["Option 1", "Option 2"]

But instead I get:

    chooseTwo: true

I have reviewed my code, and also compared with several code examples, but don't seem to find the issue. I use Material UI and also the Controller from this library.

Here is a sandbox: https://codesandbox.io/s/mui-rfhv6-forked-hrkll?file=/src/DynamicForm.js

I do, however, notice that my "value" is not set. See: 
![image](https://user-images.githubusercontent.com/3786627/123682675-92aead00-d84b-11eb-9c6f-a6440ebc22a2.png)

I have tried hard-coding value like so:

    control={<Checkbox {...field} value={option} inputRef={ref} />}

That changes to value to the expected, but I still get "true" in callback.

**Comment by simplenotezy** [comment, marked as answer]
update: just found https://github.com/react-hook-form/react-hook-form/discussions/4920 - will review and see

**Reply by simplenotezy** [reply]
Unfortunately still no solution. The example from https://github.com/react-hook-form/react-hook-form/discussions/4920 follows my approach very much (I do use Controller, however), so it was easy to compare.

I've ensured that "inputRef" is passed to the <Checkbox>. Still, I only get `true` returned.

**Reply by jorisre** [reply]
Hi @simplenotezy 
Here is a possible solution: https://codesandbox.io/s/mui-rfhv6-forked-4473m?file=/src/DynamicForm.js

I added `defaultValue={[]}` to controller and `value` to checkbox as well. I've overwritten the `onChange` prop to match your needs.

**Reply by simplenotezy** [reply]
@jorisre Thanks for your suggestion. That was a great workaround, thanks a lot for helping me out here. Really got me frustrated that I spent so much time debugging the issue, and here you come, fixing it in no-time ;)
